boot/initramfs.go

```diff
@@ -125,3 +127,57 @@ func MockInitramfsReboot(f func() error) (restore func()) {
 func InitramfsReboot() error {
 	return initramfsReboot()
 }
+
+func updatePibootKernelStatus(bl bootloader.Bootloader) error {
+	blVars, err := bl.GetBootVars("kernel_status")
+	if err != nil {
+		return err
+	}
+	curKernStatus := blVars["kernel_status"]
+	if curKernStatus == "" {
+		return nil
+	}
+
+	cmdLine, err := osutil.KernelCommandLine()
+	if err != nil {
+		return err
+	}
+	args := strings.Split(cmdLine, " ")
+	// "" would be the value for the error case
+	newStatus := ""
+	for _, arg := range args {
+		if strings.HasPrefix(arg, "kernel_status=") && curKernStatus == "try" {
```

Comment:
this code needs comments or to be clarified, it seems we consider curKernStatus only if kernel_status appears on the kernel command line? and we use the value from there? but it must be trying but we don't check it is?